File: src/46-60/52.py
import sys


def getIt(s):
    res = []
    i = 0
    while i < len(s):
        if s[i] == '[':
            t = ''
            j = i + 1
            while j < len(s) and s[j] != ']':
                t += s[j]
                j += 1
            if j != len(s):
                res.append(t)
            i = j + 1
        else:
            i += 1
    return res


def minput():
    data = sys.stdin.buffer.readline().strip()
    i = 0
    result = []
    while i < len(data):
        if data[i] <= 127:
            # ASCII 字符，一个字节处理
            result.append(chr(data[i]))
            i += 1
        else:
            # 非 ASCII 字符，两个字节处理
            if i + 1 < len(data):
                result.append(data[i:i+2].decode('utf-8', errors='ignore'))
                i += 2
            else:
                # 如果最后一个字节是非 ASCII 字符且没有足够的字节组成一个字符
                result.append(chr(data[i]))
                i += 1
    return ''.join(result)


hand = getIt(minput())
eye = getIt(minput())
mouth = getIt(minput())
od = [hand, eye, mouth, eye, hand]
n = int(minput())
for _ in range(n):
    lst = list(map(int, minput().split()))
    if len(lst) != 5 or any(lst[i] < 1 or lst[i] > len(od[i]) for i in range(5)):
        print('Are you kidding me? @\\/@')
        continue
    res = "{}({}{}{}){}".format(*(od[i][lst[i] - 1] for i in range(5)))
    print(res)

File: src/46-60/52.cpp
#include <iostream>
#include <string>

using namespace std;

int getString(string str, string a[]) {
  int k = 0;
  for (int i = 0; i < str.length(); i++) {
    if (str[i] != '[' && str[i] != ' ' && str[i] != ']') {
      a[k] += str[i];
    }
    if (str[i] == ']')
      k++;
  }
  return k;
}

bool isValid(int a, int b) {
  if (a <= b && a > 0)
    return true;
  else
    return false;
}

string s[3];
string hand[11], eye[11], mouth[11];
int main() {
  for (int i = 0; i < 3; i++) {
    getline(cin, s[i]);
  }
  int len1 = getString(s[0], hand);
  int len2 = getString(s[1], eye);
  int len3 = getString(s[2], mouth);
  int n;
  cin >> n;
  int num[n][5];
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < 5; j++) {
      cin >> num[i][j];
    }
  }
  for (int i = 0; i < n; i++) {
    int a = num[i][0];
    int b = num[i][1];
    int c = num[i][2];
    int d = num[i][3];
    int e = num[i][4];
    if (isValid(a, len1) && isValid(b, len2) && isValid(c, len3) &&
        isValid(d, len2) && isValid(e, len1)) {
      cout << hand[a - 1] << "(" << eye[b - 1] << mouth[c - 1] << eye[d - 1]
           << ")" << hand[e - 1] << endl;
    } else {
      cout << "Are you kidding me? @\\/@" << endl;
    }
  }
  return 0;
}